source/Calamari/Kubernetes/Conventions/GatherAndApplyRawYamlConvention.cs

```diff
@@ -77,12 +81,23 @@ private List<string> GroupFilesIntoDirectories(RunningDeployment deployment, str
             var directories = new List<string>();
             foreach (var (glob, idx) in globs.Select((g, i) => (g, i)))
             {
-                var files = Directory.GetFiles(deployment.CurrentDirectory, glob);
                 var directory = Path.Combine(deployment.CurrentDirectory, "grouped", idx.ToString());
-                Directory.CreateDirectory(directory);
-                foreach (var file in files)
+                fileSystem.CreateDirectory(directory);
+
+                var matcher = new Matcher();
+                matcher.AddInclude(glob);
+                var result = matcher.Execute(new DirectoryInfoWrapper(new DirectoryInfo(deployment.CurrentDirectory)));
+                foreach (var file in result.Files)
                 {
-                    fileSystem.CopyFile(file, Path.Combine(directory, Path.GetFileName(file)));
+                    var relativeFilePath = file.Path ?? file.Stem;
+                    var fullPath = Path.Combine(deployment.CurrentDirectory, relativeFilePath);
+                    var targetPath = Path.Combine(directory, relativeFilePath);
+                    var targetDirectory = Path.GetDirectoryName(targetPath);
+                    if (targetDirectory != null)
+                    {
+                        fileSystem.CreateDirectory(targetDirectory);
+                    }
```

Comment:
Can we just do this here?  It saves us a retrieval of the path info.   I'm not massively stressed either way, but depending on the number of files returned it could be a bit faster.
```suggestion
                        fileSystem.CreateDirectory(targetPath); // Only creates the directory if it doesn't exist
```

Reply:
Does this work if the path includes the file name?

Reply:
Oh right! No, I imagine it wouldn't.